modules/example/src/smithy4s/example/Launcher.scala

```diff
@@ -31,7 +31,7 @@ object Launcher extends IOApp {
     val serverRes = for {
       impl <- Resource.eval(ObjectServiceImpl.makeIO)
       docs = smithy4s.http4s.swagger.docs[IO](ObjectService)
-      service <- SimpleRestJsonBuilder.routes(impl).resource
+      service <- SimpleRestJsonBuilder.build.routes(impl).resource
```

Comment:
Hmm not sure I like the idea of building the builder like this. Seems like the ideal UX would be something like:

```
SimpleRestJsonBuilder.routes(impl).withMaxJsonArity(2048).resource
```
where the call to `withMaxJsonArity` is not necessary if you want to use the default.

We could also take the more flexible approach of:

```
SimpleRestJsonBuilder.routes(impl).withJsonCodec(myJsonCodecApi).resource
```

What do you think?

Reply:
I definitely agree,
The problem is that SimpleProtocolBuilder doesnt have access to the JsonCodecApi as its typed to CodecApi, so we cant access the existing codec to make a copy .
We can create a new instance of JsonCodecApi inside the builder

Reply:
does it even need to be typed to `JsonCodecApi`? Couldn't it be just `CodecApi`?

Reply:
@kubukoz does it make sense to allow any codec on a RestJsonBuilder ?

Reply:
hmm now that you said it out loud...